File: java/UART.java
/**
 *  Java Grinder
 *  Author: The Naken Crew
 *   Email: mike@mikekohn.net
 *     Web: http://www.naken.cc/
 * License: GPL
 *
 * Copyright 2013 by The Naken Crew
 *
 */

package cc.naken.java_grinder;

public class UART
{
  private UART()
  {
  }

  public static void init(int port, int baud_rate)
  {
  }

  public static void send(byte c)
  {
     System.out.println("Sent char " + c);
  }

  public static byte read(byte c)
  {
    return 0;
  }

  public static boolean isDataAvailable()
  {
    return false;
  }

  public static boolean isSendReady()
  {
    return false;
  }
}

File: api/uart.cxx
/**
 *  Java Grinder
 *  Author: Michael Kohn
 *   Email: mike@mikekohn.net
 *     Web: http://www.mikekohn.net/
 * License: GPL
 *
 * Copyright 2014-2015 by Michael Kohn
 *
 */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdint.h>

#include "JavaClass.h"
#include "uart.h"

#define CHECK_FUNC(funct,sig) \
  if (strcmp(#funct#sig, method_name) == 0) \
  { \
    return generator->uart_##funct(port); \
  }

#define CHECK_FUNC_CONST(funct,sig) \
  if (strcmp(#funct#sig, method_name) == 0) \
  { \
    return generator->uart_##funct(port, const_val); \
  }

int uart(JavaClass *java_class, Generator *generator, char *method_name, int port)
{
  CHECK_FUNC(init,_I)
  CHECK_FUNC(send,_I)
  CHECK_FUNC(read,)
  CHECK_FUNC(isDataAvailable,)
  CHECK_FUNC(isSendReady,)

  return -1;
}

int uart(JavaClass *java_class, Generator *generator, char *method_name, int port, int const_val)
{
  CHECK_FUNC_CONST(init,_I)

  return -1;
}



